### PR TITLE
add shared tasks and dependencies

### PR DESCRIPTION
**Pull request by josiahsprague**
This adds shared tasks for jungle-solidus sites.

- [x] compileJs
- [x] concat (css)

### CODE REVIEW

**Comment by josiahsprague** [comment]
The compileJs task should probably be updated to use the browserify task from broadway.

File: tasks/compileJs.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+var concat = require('gulp-concat');
+var gulp = require('gulp');
+var source = require('vinyl-source-stream');
+var buffer = require('vinyl-buffer');
+var browserifyify = require('broadway/tasks/browserify');
+
+function plugin(browserifyEntries, legacyScripts){
+  return(lazypipe()
+    .pipe(browserifyify, browserifyEntries)
+    .pipe(source, 'browserified.js')
```
Comment (pushred) [inline tasks/compileJs.js]: I think the filename isn't needed in this case?

Comment (pushred) [inline tasks/compileJs.js]: Nope it needed it, blew up with: 

```
Error: No path specified! Can not get relative.
    at File.Object.defineProperty.get (/Users/eric/dev/sites/bigtalkmusic.com/node_modules/vinyl-source-stream/node_modules/vinyl/index.js:153:27)
```

File: tasks/compileJs.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+var lazypipe = require('lazypipe');
+var concat = require('gulp-concat');
+var gulp = require('gulp');
+var source = require('vinyl-source-stream');
+var buffer = require('vinyl-buffer');
+var browserifyify = require('broadway/tasks/browserify');
+var browserifyEntries = require.resolve('jungle-solidus/scripts');
```
Comment (pushred) [inline tasks/compileJs.js]: Please stub this out so nobody runs into:

```
TypeError: Cannot read property 'undefined' of undefined
    at Object.module.exports (/Users/eric/dev/sites/janetjackson.com/node_modules/jungle-solidus/helpers/assets.js:8:36)
    at Object.eval (eval at <anonymous> (/Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/handlebars/lib/handlebars/compiler/compiler.js:579:23), <anonymous>:63:72)
    at /Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/handlebars/lib/handlebars/runtime.js:38:33
    at /Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/handlebars/lib/handlebars/compiler/compiler.js:1294:21
    at ExpressHandlebars._renderTemplate (/Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/express-handlebars/lib/express-handlebars.js:310:12)
    at ExpressHandlebars.<anonymous> (/Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/express-handlebars/lib/express-handlebars.js:184:21)
    at /Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/express-handlebars/node_modules/promise/lib/core.js:33:15
    at flush (/Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/express-handlebars/node_modules/promise/node_modules/asap/asap.js:27:13)
    at /Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/continuation-local-storage/node_modules/async-listener/glue.js:188:31
    at process._tickCallback (node.js:442:13)
```

All that's needed is:

```
module.exports = {};
```

#2 will address this properly, but I don't want to merge this branch in this state.

**Comment by pushred** [comment]
Can you add some handling around sections? I didn't have any defined in my config and ran into this:

```
TypeError: Cannot read property 'undefined' of undefined
    at Object.module.exports (/Users/eric/dev/sites/janetjackson.com/node_modules/jungle-solidus/helpers/assets.js:11:36)
    at Object.eval (eval at <anonymous> (/Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/handlebars/lib/handlebars/compiler/compiler.js:579:23), <anonymous>:63:72)
    at /Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/handlebars/lib/handlebars/runtime.js:38:33
    at /Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/handlebars/lib/handlebars/compiler/compiler.js:1294:21
    at ExpressHandlebars._renderTemplate (/Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/express-handlebars/lib/express-handlebars.js:310:12)
    at ExpressHandlebars.<anonymous> (/Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/express-handlebars/lib/express-handlebars.js:184:21)
    at /Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/express-handlebars/node_modules/promise/lib/core.js:33:15
    at flush (/Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/express-handlebars/node_modules/promise/node_modules/asap/asap.js:27:13)
    at /Users/eric/dev/sites/janetjackson.com/node_modules/solidus/node_modules/continuation-local-storage/node_modules/async-listener/glue.js:188:31
    at process._tickCallback (node.js:442:13)
```

Some sites may not define any sections and we shouldn't require full config before a server can be started. You can stub it out of course, but the error handling is pretty bad. It'd be easier to just make it optional.

File: tasks/compileJs.js (hunk 2 above)
Comment (pushred) [inline tasks/compileJs.js]: In fact, you don't even need any code. An empty file does the job.

**Comment by pushred** [comment]
:+1: with https://github.com/SparkartGroupInc/jungle-solidus/commit/92494b201dea562169937b2097812709fbffa1bc

**Pull request merged**